### ISSUE TITLE
Question about get_3d_box(...) function in utils/box_util.py 

### ISSUE THREAD

**Issue by cheng052** [daveredrum/ScanRefer#13]
Hi, @daveredrum,

 I have a question about the implementation of get_3d_box(...) in utils/box_util.py [https://github.com/daveredrum/ScanRefer/blob/a728f21feb73b8f491b7be50bd46d5c390d16fa7/utils/box_util.py#L290](url)

I see that you have changed the corresponding order of channels between box_size and center.

However, the get_3d_box() function is also called in function parse_predictions() in lib/ap_helper.py [https://github.com/daveredrum/ScanRefer/blob/a728f21feb73b8f491b7be50bd46d5c390d16fa7/lib/ap_helper.py#L40](url)

In parse_predictions() function, before the get_3d_box() is called, the pred_center has been transformed into pred_center_upright_camera by flip_axis_to_camera(), which changes the Y,Z order of point cloud data.

So, I'm a little confused about the get_3d_box() implementation, and hope for your explanation about the change compared with original votenet.  

Can you check this problem? It will be of great help.

**Comment by daveredrum** [comment]
Hi @cheng052,

Thanks for pointing this out! This issue seems to be related to #11 (NMS-related). As we set Z-axis as our up-axis in our project, we need to flip Y and Z axes in the `get_3d_box()` in the original VoteNet. This modification enables direct downstream application in this project, such as visualizing the bounding boxes or dynamic label assignment in the `loss_helper.py`. `flip_axis_to_camera()` seems to be a problamatic legacy from VoteNet, since our project doesn't need NMS while VoteNet detections do.

In a nutshell, the issue with NMS won't affect the localization performance, but it will for the detections. I'll look into this issue to make sure the detections with NMS are also bug-free.
